src/kubernetes/api_watcher.rs

```diff
@@ -77,12 +86,7 @@ where
         let status = response.status();
         if status != StatusCode::OK {
             let source = invocation::Error::BadStatus { status };
-            let err = if status == StatusCode::GONE {
-                watcher::invocation::Error::desync(source)
```

Comment:
Do we not want to handle `desync` events specifically anymore? Or do they end up as a `hyper::Error` and thus mapped to `Recoverable`.

Reply:
Related to this,

if they _are_ part of `Recoverable`, shouldn't we still reset the internal state as was previously done?

Reply:
Agreed. It seems like, specifically on desync, we should be clearing the internal state (looking at https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes).

My opinion is to re-add the existing specific handling of `desync` events and retry everything else.

Reply:
Upon closer inspection the desync events are actually also handled lower down in `api_watcher.rs` by reading the status code from the body. I'm guessing k8s still returns a JSON payload with the 410 that has the status within the body. However, https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes, specifically mentions handling HTTP 410 responses, so I readded that logic.

> if they are part of Recoverable, shouldn't we still reset the internal state as was previously done?

I don't think we want to clear the state if there was a network blip. I think it's better to just keep running with the stale state. I could be convinced otherwise though.